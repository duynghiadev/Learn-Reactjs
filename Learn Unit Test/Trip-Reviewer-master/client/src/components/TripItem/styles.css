.trip-item-container {
  border-bottom: 4px solid #f1f1f1;
  box-sizing: border-box;
  padding: 10px 20px;
  display: block;
  background: #f5f5f578;
}
.trip-item-container:hover {
  background: #f5f5f5;
}
.trip-item-container .header h2 {
  font-size: 20px;
  color: #686868;
  margin: 4px 0px 0px 0px;
  font-weight: 300;
}
.trip-item-container .items {
  font-size: 13px;
}
.trip-item-container .items div {
  margin-bottom: 8px;
}
.trip-item-container .country {
  margin-top: 5px;
  font-size: 15px;
  color: #5f5f5f;
  font-weight: 700;
}
.trip-item-container .bubble {
  background: #e91e63;
  display: inline-block;
  padding: 3px 5px;
  border-radius: 3px;
  font-size: 11px;
  margin-right: 5px;
  color: #fff;
}
