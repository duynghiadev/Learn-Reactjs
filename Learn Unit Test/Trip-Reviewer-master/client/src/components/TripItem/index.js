import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { TRIPS } from '../../constants/routes';
import { TYPES } from '../Common/DetailsLabel/utils';
import DetailsLabel from '../Common/DetailsLabel';

import './styles.css';

const TripItem = ({ _id, title, country, expences, duration, rating }) => (
  <Link to={`${TRIPS}/${_id}`} className="trip-item-container">
    <div className="header">
      <h2>{title}</h2>
    </div>
    <div className="items">
      <div className="country">{country}</div>

      <DetailsLabel type={TYPES.expences} value={expences}>
        <strong>Expenses:</strong>
        {` $${expences}`}
      </DetailsLabel>

      <DetailsLabel type={TYPES.duration} value={duration}>
        <strong>Duration:</strong>
        {duration === 1 ? ' 1 day' : ` ${duration} days`}
      </DetailsLabel>

      <DetailsLabel type={TYPES.rating} value={rating}>
        <strong>Rating:</strong>
        {` ${rating} / 5`}
      </DetailsLabel>
    </div>
  </Link>
);

TripItem.propTypes = {
  _id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired,
  expences: PropTypes.number.isRequired,
  duration: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
};

export default TripItem;
