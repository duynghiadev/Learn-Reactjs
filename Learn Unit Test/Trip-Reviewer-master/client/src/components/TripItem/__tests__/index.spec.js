import React from 'react';
import { create } from 'react-test-renderer';

import TripItem from '..';

jest.mock('react-router-dom', () => ({ Link: 'Link ' }));
jest.mock('../../Common/DetailsLabel', () => 'DetailsLabel');

const mockComponent = props => <TripItem {...props} />;

describe('<TripItem />', () => {
  it('should render component with duration 1', () => {
    const props = {
      _id: '_id',
      country: 'country',
      expences: 'expences',
      duration: 1,
      rating: 'rating',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with duration 10', () => {
    const props = {
      _id: '_id',
      country: 'country',
      expences: 'expences',
      duration: 10,
      rating: 'rating',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
