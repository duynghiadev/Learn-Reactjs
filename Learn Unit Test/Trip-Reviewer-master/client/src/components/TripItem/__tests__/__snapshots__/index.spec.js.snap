// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripItem /> should render component with duration 1 1`] = `
<Link 
  className="trip-item-container"
  to="/trips/_id"
>
  <div
    className="header"
  >
    <h2 />
  </div>
  <div
    className="items"
  >
    <div
      className="country"
    >
      country
    </div>
    <DetailsLabel
      type="EXPENCES"
      value="expences"
    >
      <strong>
        Expenses:
      </strong>
       $expences
    </DetailsLabel>
    <DetailsLabel
      type="DURATION"
      value={1}
    >
      <strong>
        Duration:
      </strong>
       1 day
    </DetailsLabel>
    <DetailsLabel
      type="RATING"
      value="rating"
    >
      <strong>
        Rating:
      </strong>
       rating / 5
    </DetailsLabel>
  </div>
</Link >
`;

exports[`<TripItem /> should render component with duration 10 1`] = `
<Link 
  className="trip-item-container"
  to="/trips/_id"
>
  <div
    className="header"
  >
    <h2 />
  </div>
  <div
    className="items"
  >
    <div
      className="country"
    >
      country
    </div>
    <DetailsLabel
      type="EXPENCES"
      value="expences"
    >
      <strong>
        Expenses:
      </strong>
       $expences
    </DetailsLabel>
    <DetailsLabel
      type="DURATION"
      value={10}
    >
      <strong>
        Duration:
      </strong>
       10 days
    </DetailsLabel>
    <DetailsLabel
      type="RATING"
      value="rating"
    >
      <strong>
        Rating:
      </strong>
       rating / 5
    </DetailsLabel>
  </div>
</Link >
`;
